pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -221,6 +222,23 @@ func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl
 		return ctrl.Result{Requeue: true}, nil
 	}
 
+	var statusMessage string
+	successStatusMessage := "Deployment %s/%s have correct number of AvailableReplicas which is %s"
+	failStatusMessage := "Deployment %s/%s have incorrect number of AvailableReplicas, have %s but should be %s"
+	if found.Status.AvailableReplicas != size {
+		statusMessage = fmt.Sprintf(failStatusMessage, {{ lower .Resource.Kind }}.Name, {{ lower .Resource.Kind }}.Namespace, found.Status.AvailableReplicas, size)
+	} else {
+		statusMessage = fmt.Sprintf(successStatusMessage, {{ lower .Resource.Kind }}.Name, {{ lower .Resource.Kind }}.Namespace, size)
+	}
+	if !reflect.DeepEqual(statusMessage, {{ lower .Resource.Kind }}.Status.DeploymentStatus){
+		{{ lower .Resource.Kind }}.Status.DeploymentStatus = statusMessage
+		err := r.Status().Update(ctx, {{ lower .Resource.Kind }})
+		if err != nil {
+			log.Error(err, "Failed to update {{ lower .Resource.Kind }} status")
+			return ctrl.Result{}, err
+		}
+	}
+
```

Comment:
Hi @NikhilSharmaWe,  

About the status that would be like something: https://github.com/kubernetes-sigs/kubebuilder/compare/master...camilamacedo86:kubebuilder:test-br?expand=1

See that the convention defines how we should fille the conditional one and we need to keep a track of the transaction. You can check the description of its field/spec in the code implementation here: https://github.com/kubernetes-sigs/kubebuilder/pull/2816/files#r919781621 (were we are commenting what should be added to each case scenario)

- See the example (https://kpt.dev/reference/schema/crd-status-convention/) it might help you get a better idea.
- Also, ensure that you check the documentation: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#spec-and-status